packages/legacy/core/App/screens/Connection.tsx

```diff
@@ -147,54 +121,45 @@ const Connection: React.FC<ConnectionProps> = ({ navigation, route }) => {
   }, [state.shouldShowDelayMessage])
 
   useEffect(() => {
-    // for non-connectionless, invalid connections stay in the below state
-    if (connection && connection.state === DidExchangeState.RequestSent) {
-      logger?.info('Connection: Skipping invalid connection state')
-
+    if (!state.notificationRecord || !state.isVisible) {
       return
     }
 
-    if (
-      !connectionId &&
-      !oobRecord &&
-      !goalCode &&
-      state.notificationRecord &&
-      state.notificationRecord.state === 'request-received'
```

Comment:
I think it can go. I does not match any conditions in @cvarjao hackmd doc and I can't think of any use cases. Let's ax it for now and if something shows up in 1.0.19 we'll add it back with documentation as to why. Also, useNotifications won't return any in that state will it?

Reply:
I think it's possible for BC Wallet's `useNotifications` implementation to include proofs in state `Done` and in state `PresentationReceived`. It might be better for us to fix that logic in our implementation of that hook rather than in this screen though. So I'm good with merging this